src/openforms/formio/dynamic_config/tests/test_date_component.py

```diff
@@ -0,0 +1,265 @@
+from copy import deepcopy
+from datetime import datetime
+from typing import Any, Dict
+
+from django.test import SimpleTestCase
+from django.utils import timezone
+
+from freezegun import freeze_time
+from rest_framework.test import APIRequestFactory
+
+from openforms.submissions.tests.factories import SubmissionFactory
+from openforms.variables.service import get_static_variables
+
+from ...dynamic_config.date import FormioDateComponent
+from ...service import get_dynamic_configuration
+from ...typing import Component
+
+request_factory = APIRequestFactory()
+
+
+class DynamicDateConfigurationTests(SimpleTestCase):
+    @staticmethod
+    def _get_dynamic_config(
+        component: Component, variables: Dict[str, Any]
+    ) -> FormioDateComponent:
+        configuration = {"components": [component]}
+        request = request_factory.get("/irrelevant")
+        submission = SubmissionFactory.build()
+        static_vars = get_static_variables(submission=None)  # don't do queries
+        variables.update({var.key: var.initial_value for var in static_vars})
+        new_configuration = get_dynamic_configuration(
+            configuration, request=request, submission=submission, data=variables
+        )
+        return new_configuration["components"][0]
+
+    def test_legacy_configuration_still_works(self):
+        # legacy configuration = without the openForms.minDate keys etc.
+        component = {
+            "type": "date",
+            "key": "aDate",
+            "datePicker": {
+                "minDate": None,
+                "maxDate": "2022-09-08T00:00:00+02:00",
```

Comment:
I assume you checked that formio uses this (timezone aware) date and not regular dates? It's almost worth writing a test to check that with a JS test.

Reply:
yep, this is what ends up in the JSON when you pick an explicit min/max date - a full datetime string.